iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -38,101 +38,113 @@ internal sealed class HttpClientHelper : IHttpClientHelper
         private readonly bool _usingX509ClientCert;
         private HttpClient _httpClientObj;
-        private HttpClientHandler _httpClientHandler;
+        private HttpMessageHandler _httpMessageHandler;
         private bool _isDisposed;
         private readonly ProductInfo _productInfo;
         private readonly bool _isClientPrimaryTransportHandler;
 
-        // These default values are consistent with Azure.Core default values:
-        // https://github.com/Azure/azure-sdk-for-net/blob/7e3cf643977591e9041f4c628fd4d28237398e0b/sdk/core/Azure.Core/src/Pipeline/ServicePointHelpers.cs#L28
-        private const int DefaultMaxConnectionsPerServer = 50;
-
-        // How long, in milliseconds, a given cached TCP connection created by this client's HTTP layer will live before being closed.
-        // If this value is set to any negative value, the connection lease will be infinite. If this value is set to 0, then the TCP connection will close after
-        // each HTTP request and a new TCP connection will be opened upon the next request.
-        //
-        // By closing cached TCP connections and opening a new one upon the next request, the underlying HTTP client has a chance to do a DNS lookup
-        // to validate that it will send the requests to the correct IP address. While it is atypical for a given IoT hub to change its IP address, it does
-        // happen when a given IoT hub fails over into a different region.
-        //
-        // This default value is consistent with the default value used in Azure.Core
-        // https://github.com/Azure/azure-sdk-for-net/blob/7e3cf643977591e9041f4c628fd4d28237398e0b/sdk/core/Azure.Core/src/Pipeline/ServicePointHelpers.cs#L29
-        private static readonly TimeSpan s_defaultConnectionLeaseTimeout = TimeSpan.FromMinutes(5);
-
         public HttpClientHelper(
             Uri baseAddress,
             IAuthorizationProvider authenticationHeaderProvider,
             IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> defaultErrorMapping,
             TimeSpan timeout,
-            Action<HttpClient> preRequestActionForAllRequests,
-            X509Certificate2 clientCert,
-            HttpClientHandler httpClientHandler,
+            Http1TransportSettings transportSettings,
             ProductInfo productInfo,
             IWebProxy proxy,
             bool isClientPrimaryTransportHandler = false)
         {
             _baseAddress = baseAddress;
             _authenticationHeaderProvider = authenticationHeaderProvider;
             _defaultErrorMapping = new ReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>(defaultErrorMapping);
+            _productInfo = productInfo;
+            _isClientPrimaryTransportHandler = isClientPrimaryTransportHandler;
+
+            if (transportSettings.HttpClient != null)
+            {
+                _httpClientObj = transportSettings.HttpClient;
+                return;
```

Comment:
I thought _httpMessageHandler  initializing was missing but turns out that it isn't required to be set for this case since it is used only as a local variable when creating an HttpClient. We should remove it from class variables and make it a local variable.

Reply:
Good point